apps/src/code-studio/components/progress/LessonLockDialog/LessonLockDataApi.js

```diff
@@ -0,0 +1,104 @@
+import {useMemo} from 'react';
+import _ from 'lodash';
+import {makeEnum} from '@cdo/apps/utils';
+import {useFetch} from '@cdo/apps/util/useFetch';
+
+export const LockStatus = makeEnum('Locked', 'Editable', 'ReadonlyAnswers');
+
+/**
+ * @typedef {Object} UserLockState
+ * @property {string} name name of student
+ * @property {LockStatus} lockStatus lock status (corresponds to the radio buttons in the dialog)
+ * @property {Object} userLevelData opaque user_level data sent by server used to identify a user's lock data
+ */
+
+/**
+ * @typedef {UserLockState[]} LockState array of UserLockState objects, usually
+ *    representing the lock state for the students in a one section
+ */
+
+/**
+ * Retrieves the lock state from the server and extracts the data needed by
+ * the LeesonLockDialog.
+ * @param {number} unitId unit id
+ * @param {number} lessonId lesson id
+ * @param {number} sectionId section id
+ * @returns {{loading: boolean, serverLockState: LockState}}
+ */
+export function useGetLockState(unitId, lessonId, sectionId) {
+  const {loading, data} = useFetch(`/api/lock_status?script_id=${unitId}`);
+
+  const serverLockState = useMemo(
+    () => extractLockData(data, sectionId, lessonId),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
For my understanding, why does linting need to be disabled here?

Reply:
I looked into this, and James seemed right that there must be an error in exhaustive-deps lint rule because there are required deps included here that it thinks aren't necessary.